### ISSUE TITLE
Missing source code attribution

### ISSUE THREAD

**Issue by asanetargoss** [ZeroPoints/Ensoulomancy#7]
Hello there! It has come to my attention that this mod uses a significant amount of code from Metamorph. Metamorph is MIT licensed, so you're within your right to do so. however, the MIT license also requires you distribute the license along with anything that uses Metamorph's code. You can find the license [here](https://github.com/mchorse/metamorph/blob/master/LICENSE.md). I'm sure @mchorse will appreciate being credited for his work.

I also will just put it out there, I find it odd that you don't just have Metamorph as a dependency instead. It would save you the trouble of having to port over bugfixes from Metamorph in the future, and the Metamorph repo is always open to PRs if there is some functionality you feel is missing.

**Comment by ChickenMobile** [comment]
It is true that a lot of the Metamorph code is used and we will credit mchorse. I appreciate their work, by far one of the better readable (and commented) code out there.

Also a lot of the Metamorph code has been stripped out as it is unneeded and will be highly modified in the future. If anything: currently most of the code is kept for learning reasons.

Thank you for bringing this to our attention.

**Comment by ZeroPoints** [comment]
Included Metamorph License at base of project.
https://github.com/ZeroPoints/Ensoulomancy/blob/master/METAMOPRH-LICENSE.md
Updated Ensoulomancy License as we hadn't included it into the github project as of yet.
https://github.com/ZeroPoints/Ensoulomancy/blob/master/LICENSE.md
Updated Read Me to reflect credits where credits due
https://github.com/ZeroPoints/Ensoulomancy/blob/master/README.md

**Comment by ZeroPoints** [comment]
Aaaahhh didn't want to close it. This is our first Java project and our first minecraft mod. So we are also going to look into seeing how "dependencies" works and see what we have so far changed from Metamorph and see if we can just include yours and override for the extra functionality/less functionality that we want to implement in our mod. We don't want to take away from your mods purpose for morphing. Feel free to message us if you have more issues.

**Comment by mchorse** [comment]
Thanks for notifying me @asanetargoss!

Yeah, I don't have problems with you using my code as long as you give me credit in README's and `mcmod.info`. Thanks! I try to keep the code tidy and documented, but that's not always possible :joy: 

Also, I would really appreciate if you would put somewhere "incompatible with **Metamorph** mod", because I feel like unaware users might try using Metamorph with Ensoulomancy and this might cause the *notorious* incompatibility with MPM (basically screen jiggle due to both Metamorph and MPM modifying `eyeHeight`). So yeah, if you would add that note, it would save me a lot of headache from users reporting same stupid issue over and over again. Thanks. :smile: :man_facepalming:

**Comment by ZeroPoints** [comment]
Hahah interesting screen jiggle issue.

**Comment by mchorse** [comment]
Yeah, basically that's what the mod option `Disable POV` was added for.

**Comment by ZeroPoints** [comment]
Would you like your curse name or gitname as credit info for mcmod.info. Or just your mod name?

**Comment by mchorse** [comment]
`McHorse` (or `mchorse`) would be fine. Unfortunately, I couldn't change curse name to McHorse when I was uploading my first mod there since it was directly tied to username on Minecraft Forums.

I wrote most of the code myself, but @asanetargoss contributed a lot with PRs and issues in the past :slightly_smiling_face:

**Comment by ZeroPoints** [comment]
Got you both in there mate :D 
https://github.com/ZeroPoints/Ensoulomancy/blob/master/src/main/resources/mcmod.info

**Comment by mchorse** [comment]
Awesome! I'm very appreciate it. :slightly_smiling_face:

**Comment by ZeroPoints** [comment]
Ok. Finally figured out how to include dependencies. Stripped out the metamorph code to include your jar as a dependency and we are gonna go on from there. I've hopefully done it correctly in all the places required.
